src/libNode/FinalBlockProcessing.cpp

```diff
@@ -1254,25 +1258,32 @@ bool Node::ProcessFinalBlock(const vector<unsigned char>& message,
     {
         LOG_GENERAL(INFO, "isVacuousEpoch now");
 
-        if (AccountStore::GetInstance().UpdateStateTrieAll()
-            && !CheckStateRoot(txBlock))
+        if (AccountStore::GetInstance().UpdateStateTrieAll())
         {
+            if (!CheckStateRoot(txBlock))
+            {
 #ifndef IS_LOOKUP_NODE
-            m_mediator.m_isConnectedToNetwork = false;
-            this->Init();
-            this->Prepare(true);
-            this->StartSynchronization();
+                RejoinAsNormal();
 #endif // IS_LOOKUP_NODE
-            return false;
+                return false;
+            }
+            else
+            {
+                StoreState();
+                BlockStorage::GetBlockStorage().PutMetadata(
+                    MetaType::DSINCOMPLETED, {'0'});
+#ifndef IS_LOOKUP_NODE
+                BlockStorage::GetBlockStorage().PopFrontTxBodyDB();
+#else // IS_LOOKUP_NODE
+                BlockStorage::GetBlockStorage().ResetDB(
+                    BlockStorage::TX_BODY_TMP);
+#endif // IS_LOOKUP_NODE
+            }
         }
         else
```

Comment:
This "else" case could be moved before "if" case, handle the error case first could reduce the nested structure

Reply:
ok